osu.Game/Overlays/Settings/Sections/Input/KeyBindingRow.cs

```diff
@@ -470,12 +477,19 @@ public KeyButton(RealmKeyBinding keyBinding)
                         Margin = new MarginPadding(5),
                         Anchor = Anchor.Centre,
                         Origin = Anchor.Centre,
-                        Text = keyBinding.KeyCombination.ReadableString(),
                     },
                     new HoverSounds()
                 };
             }
 
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                readableKeyCombinationProvider.KeymapChanged += updateKeyCombinationText;
```

Comment:
99.99% sure this is not safe as the events are going to be fired from the input thread. feels like something that should be handled at a framework level potentially? probably only saved by the fact that updating `Text` on non-update thread is not safe but also doesn't throw..?